seaborn/rcmod.py

```diff
@@ -460,6 +433,35 @@ def set_context(context=None, font_scale=1, rc=None):
     mpl.rcParams.update(context_object)
 
 
+class _StyleOrContext(dict):
+    def __enter__(self):
+        rc = mpl.rcParams
+        self._orig = {k: rc[k] for k in self._keys}
+        type(self)._set(self)
```

Comment:
Ah I see, it's because you're calling a a non-class function stored as an attribute. Hm. I think that the `staticmethod` approach is a little bit more clear.